R/donor_control.R

```diff
@@ -0,0 +1,83 @@
+
+# This file contains methods to monitor and modify set of donor units
+# in augmented synthetic control method
+
+
+#' Return a summary data frame donor units used in the model, along
+#' with their RMSPEs and synthetic weights
+#'
+#' If the augsynth object does not have permutation-based inference
+#' results, the function will call that form of inference, in order to
+#' calculate the RMSPEs for each donor unit in turn.
+#'
+#' @param augsynth Augsynth object to be plotted
+#'
+#' @export
+donor_table <- function(augsynth) {
+
+  if (is.null(augsynth$results)) {
+    augsynth <- add_inference(augsynth, inf_type = 'permutation')
+  } else if ((!augsynth$results$inf_type %in% c("permutation", "permutaton_rstat")) | is.null(augsynth$results)) {
+    augsynth <- add_inference(augsynth, inf_type = 'permutation')
+  }
+
+  trt_index <- which(augsynth$data$trt == 1)
+  unit_var <- augsynth$unit_var
+  RMSPEs <- augsynth$results$permutations$placebo_dist %>%
+    select(!!unit_var, RMSPE) %>%
+    distinct()
+  donor_df <- RMSPEs %>% filter(!!as.name(unit_var) != augsynth$trt_unit) %>%
+    mutate(synth_weight = as.numeric( augsynth$weights ) )
+
+  return(donor_df)
+
+}
+
+
+#' Return a new augsynth object with specified donor units removed
+#' @param augsynth Augsynth object to be plotted
+#' @param drop Drop donor units, based on pre-treatment RMSPE or unit name(s)
```

Comment:
`drop` can be a numeric but it can also be a vector of unit names. For example, in the tobacco case, you can drop "AZ" as a donor unit, regardless of its RMPSE. We've expanded the description of the `update_augsynth()` function to reflect this.